src/components/molecules/EventModal/EventModal.tsx

```diff
@@ -64,10 +69,24 @@ const EventModal = ({
     setShowEventModal(false);
   }
 
+  const onCloseModal = () => {
+    if (inputFieldText) {
+      setUnsavedEvents(prevState => ({
+        ...prevState,
+        [selectedDate]: inputFieldText,
+      }));
+    } else if (unsavedEvents.hasOwnProperty(selectedDate)) {
+      const copiedUnsavedEvents = {...unsavedEvents};
+      delete copiedUnsavedEvents[selectedDate];
+      setUnsavedEvents(_ => ({...copiedUnsavedEvents}));
```

Comment:
**Question:** Could this just be `setUnsavedEvents(() => ({...copiedUnsavedEvents}));`?

Reply:
This is another PR that was moved over from Gitlab, I'm not sure who the original owner was if you'd like to take ownership and get it merged that would be great!